### PR TITLE
[DURACOM-71] Mark 'dc-qualifier' as optional

### LINKED ISSUE #8386: Optional element in submission forms
In `submission-forms.xml` there is an optional element `<dc-qualifier>`. According to its DTD this element is mandatory, and it should be marked as optional.

It is sufficient to replace this line in `submission-forms.dtd`
```xml
<!ELEMENT dc-qualifier (#PCDATA) >
```
with:
```xml
<!ELEMENT dc-qualifier (#PCDATA)? >
```

The question mark sets the element as optional.

### PR DESCRIPTION
**Pull request by davide-negretti**
## References
* Fixes #8386

- [x] My PR is small in size (e.g. less than 1,000 lines of code, not including comments & integration tests). Exceptions may be made if previously agreed upon.
- [x] My PR passes Checkstyle validation based on the [Code Style Guide](https://wiki.lyrasis.org/display/DSPACE/Code+Style+Guide).
- [x] My PR includes Javadoc for _all new (or modified) public methods and classes_. It also includes Javadoc for large or complex private methods.
- [x] My PR passes all tests and includes new/updated Unit or Integration Tests based on the [Code Testing Guide](https://wiki.lyrasis.org/display/DSPACE/Code+Testing+Guide).
- [x] If my PR includes new, third-party dependencies (in any `pom.xml`), I've made sure their licenses align with the [DSpace BSD License](https://github.com/DSpace/DSpace/blob/main/LICENSE) based on the [Licensing of Contributions](https://wiki.lyrasis.org/display/DSPACE/Code+Contribution+Guidelines#CodeContributionGuidelines-LicensingofContributions) documentation.
- [x] If my PR modifies the REST API, I've linked to the REST Contract page (or open PR) related to this change.

### CODE REVIEW

**Comment by tdonohue** [comment]
@davide-negretti : It appears that, unfortunately, this tiny change results in test failures (the test failures are saying the DTD is now invalid it appears).

File: dspace/config/submission-forms.dtd [hunk 1]
```diff
@@ -11,7 +11,7 @@
  <!ELEMENT field (dc-schema, dc-element, dc-qualifier?, language?, repeatable?, label, style?, type-bind?, input-type, hint, required?, regex?, vocabulary?, visibility?, readonly?) >
  <!ELEMENT dc-schema (#PCDATA) >
  <!ELEMENT dc-element (#PCDATA) >
- <!ELEMENT dc-qualifier (#PCDATA) >
+ <!ELEMENT dc-qualifier (#PCDATA)? >
```
Comment (mwoodiupui) [inline dspace/config/submission-forms.dtd:14]: Question mark should be *inside* the parentheses.

**Review by tdonohue — CHANGES_REQUESTED** [review]
@davide-negretti and @mwoodiupui : This is still failing tests...errors still say the DTD is invalid: 
```
org.dspace.app.util.DCInputsReaderException: Error creating submission forms: org.xml.sax.SAXParseException; systemId: file:/home/runner/work/DSpace/DSpace/dspace-api/target/testing/dspace/config/submission-forms.dtd; lineNumber: 14; columnNumber: 33; A ')' is required in the declaration of element type "dc-qualifier".
```

I've not had a chance to dig into this further, but I wanted to note that this obviously cannot move forward until we get tests functioning without errors.

**Comment by mwoodiupui** [comment]
I read more carefully this time.  I don't think that `#PCDATA` is treated as a child *element*, so perhaps it can't be optional.  I'm not sure what it would mean to say that an element might or might not have content.  I think an element can either be defined as `EMPTY`, or defined to always have content (which might be of zero length).

But the `field` element itself marks the `dc-qualifier` element as an optional child.  So the unpatched DTD should mean that you can entirely omit `dc-qualifier` within `field`, but if it is coded then it has content.  (We need to check for the special case of zero-length content, which should (I think) be treated as the element being omitted.)

I think that we need a more precise statement of the problem.  What did you want to write in the XML, and what happens when you do?  Why is that wrong?

I apologize for sowing confusion with my previous comment.

**Comment by mwoodiupui** [comment]
Tangent:  if these children had instead been attributes, we wouldn't be having this discussion.  Only `hint`, `label` and `regex` have free-form character content; the rest are tokens or Boolean.

**Comment by tdonohue** [comment]
@davide-negretti : Per discussion in last week's meeting, several of us feel this PR is not necessary as it appears the `dc-qualifier` field is already flagged as optional (`dc-qualifier?`) on this line of the current DTD: https://github.com/DSpace/DSpace/blob/main/dspace/config/submission-forms.dtd#L11

So, it's unclear why the current DTD is insufficient and why the ticket #8386 was logged in the first place.  Is there an issue you are encountering with the current DTD that we aren't seeing?

**Comment by davide-negretti** [comment]
@tdonohue I'm sorry I haven't replied to your messages, I've been on vacation. I proposed this fix with the only purpose to avoid warnings with some IDEs, however it is not necessary, and this PR can be declined.

**Pull request closed without merging**